libraries/botbuilder/src/botFrameworkAdapter.ts

```diff
@@ -1148,39 +1136,41 @@ export class BotFrameworkAdapter extends BotAdapter implements IUserTokenProvide
     /**
      * Process the initial request to establish a long lived connection via a streaming server.
      * @param req The connection request.
-     * @param res The response sent on error or connection termination.
-     * @param logic The logic that will handle incoming requests.
+     * @param socket The raw socket connection between the bot (server) and channel/caller (client).
+     * @param head The first packet of the upgraded stream.
+     * @param logic The logic that handles incoming streaming requests for the lifetime of the WebSocket connection.
      */
-    public async useWebSocket(req: WebRequest, res: WebResponse, logic: (context: TurnContext) => Promise<any>): Promise<void> {   
+    public async useWebSocket(req: IncomingMessage, socket: Socket, head: Buffer, logic: (context: TurnContext) => Promise<any>): Promise<void> {
```

Comment:
This is adding a public type dependency on Socket and Buffer. This means if there is a breaking change to these APIs between Node.js versions, we may run into type issues.

If we want to isolate ourselves from this, we can create interfaces which are compatible with the Socket and Buffer types.

Reply:
Our libraries actually don't use the `head`, we just pass it through to `ws`. 

Do we still need an interface for `head`/Buffer or should I just change the type to `any`?

Reply:
We can keep `Buffer` for now, and always swap to a compat interface or `any` later. We can't switch from `any` to anything else without it being breaking, so we should avoid that unless we're out of options.